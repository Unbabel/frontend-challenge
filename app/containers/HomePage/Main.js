import styled from 'styled-components';

import Section from 'components/Container/Section';

export default styled.main`
  height: auto;

  > ${Section} {
    width: auto;
    padding: 1.5rem;
    display: flex;
    justify-content: center;
    align-items: flex-start;
  }
`;
