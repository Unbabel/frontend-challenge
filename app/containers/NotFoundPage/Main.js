import styled from 'styled-components';

import Section from 'components/Container/Section';

export default styled.main`
  height: 50vh;

  > ${Section} {
    width: 100%;
    padding: 1.5rem;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    box-sizing: border-box;

    > span {
      font-size: 12rem;
      font-weight: bolder;
      color: ${props => props.theme.greyLight};

      & ~ span {
        font-size: 2rem;
      }
    }
  }
`;
