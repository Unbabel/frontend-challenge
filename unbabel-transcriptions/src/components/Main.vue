<script>
import axios from 'axios';
import { API_URL } from '../api/url';

export default {
  data() {
    return { transcriptions: null };
  },
  methods: {
    async getAllTranscriptions() {
      const response = await axios.get(API_URL);
      const data = await response.data;
      this.transcriptions = data;
    },
  },
  mounted() {
    this.getAllTranscriptions();
  },
};
</script>

<template>
  <section>
    <div class="container">
      <header class="container__header">
        <h1>Transcriptions</h1>
        <div class="container__header__actions">
          <span
            ><img
              src="../assets/upload.svg"
              alt="Upload icon, an arrow pointing up"
          /></span>
          <span
            ><img
              src="../assets/fetch-document.svg"
              alt="An icon with a plus sign, inside a square"
          /></span>
        </div>
      </header>
      <hr />
      <main class="container__main">
        <div
          class="container__main__card"
          v-for="transcription in transcriptions"
          :key="transcription.id"
        >
          <div class="sup">
            <div class="initial">
              <div class="intital__profile">
                <input type="checkbox" name="voice" />
                <img
                  src="../assets/person.svg"
                  alt="Blue ball representing a human face"
                />
              </div>
              <span class="container__main__card__title">{{
                transcription.voice
              }}</span>
            </div>
            <button class="container__main__card__button">
              <img src="../assets/delete.svg" alt="Trash can Icon" />
            </button>
          </div>
          <br />
          <div class="inf">
            <span class="container__main__card__title__paragraph">{{
              transcription.text
            }}</span>
          </div>
        </div>

        <div class="container__main__action">
          <button>
            <img
              src="../assets/add-row.svg"
              alt="Round button with a plus sign inside"
            />
          </button>
        </div>
      </main>
    </div>
  </section>
</template>

<style scoped>
section {
  display: flex;
  justify-content: center;
}

.container {
  display: flex;
  justify-content: center;
  flex-direction: column;
  margin: 30px 30px;
  width: 100%;
  max-width: 710px;
  border: 2px solid lightgrey;
  background-color: white;
}

.container__header {
  display: flex;
  justify-content: space-between;
  padding: 5px 10px;
  align-items: center;
  background-color: white;
  border-bottom: 2px solid lightgray;
  font-family: 'Montserrat', sans-serif;
}

.container__header__actions {
  width: 100%;
  max-width: 60px;
  display: flex;
  justify-content: space-between;
}

.container__main {
  padding: 10px;
}

.container__main__card {
  border: 1px solid lightgray;
  background-color: white;
  padding: 12px 12px;
}

.container__main__card__title {
  font-size: 16px;
  font-family: 'Montserrat', sans-serif;
  font-weight: bold;
}

.container__main__card__title__paragraph {
  font-family: 'Open Sans', sans-serif;
}

.container__main__card > .sup {
  display: flex;
  justify-content: space-between;
}

.container__main__card > .inf {
  margin-left: 50px;
}

.sup:hover > .container__main__card__button {
  display: block;
}
.container__main__card__button {
  display: none;
}

.container__main__card > .sup > .initial {
  display: flex;
  align-items: center;
}

.container__main__card > .sup > .initial > .intital__profile {
  display: inline-flex;
}

.container__main__card > .sup > .initial > span {
  margin-left: 10px;
}

.container__main__card__button {
  border: none;
  background-color: transparent;
  cursor: pointer;
}

.container__main__action {
  display: flex;
  justify-content: center;
  margin-top: 10px;
}

.container__main__action > button {
  border: none;
  background-color: transparent;
  cursor: pointer;
}
</style>
