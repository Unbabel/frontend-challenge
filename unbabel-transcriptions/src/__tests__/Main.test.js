import { mount } from '@vue/test-utils';
import Main from '../components/Main.vue';

describe('Main Component', () => {
  it('should render Main without crash', () => {
    const wrapper = mount(Main);

    expect(wrapper.text()).toContain('Transcriptions');
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render Loading', () => {
    const wrapper = mount(Main, {
      data() {
        return {
          loading: true,
        };
      },
    });

    expect(wrapper.text()).toContain('Loading...');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
