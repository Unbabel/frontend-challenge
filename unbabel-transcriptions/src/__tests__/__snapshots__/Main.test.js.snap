// Vitest Snapshot v1

exports[`Main Component > should render Main without crash 1`] = `
"<section data-v-3fa90f43=\\"\\">
  <div class=\\"container\\" data-v-3fa90f43=\\"\\">
    <header class=\\"container__header\\" data-v-c970699f=\\"\\" data-v-3fa90f43=\\"\\">
      <h1 data-v-c970699f=\\"\\">Transcriptions</h1>
      <div class=\\"container__header__actions\\" data-v-c970699f=\\"\\"><button class=\\"container__header__actions__button\\" data-v-c970699f=\\"\\"><img src=\\"/src/assets/upload.svg\\" alt=\\"Upload icon, an arrow pointing up\\" data-v-c970699f=\\"\\"></button><button class=\\"container__header__actions__button\\" data-v-c970699f=\\"\\"><img src=\\"/src/assets/fetch-document.svg\\" alt=\\"An icon with a plus sign, inside a square\\" data-v-c970699f=\\"\\"></button></div>
    </header>
    <main class=\\"container__main\\" data-v-3fa90f43=\\"\\">
      <!--v-if-->
      <p class=\\"container__main__message\\" data-v-3fa90f43=\\"\\"> Click on <img src=\\"/src/assets/fetch-document.svg\\" alt=\\"An icon with a plus sign, inside a square\\" data-v-3fa90f43=\\"\\"> to display the transcriptions </p>
      <!--v-if-->
    </main>
  </div>
</section>"
`;
